src/lib/services/proposalService.ts

```diff
@@ -0,0 +1,168 @@
+          proposalId: proposal.proposal_id,
+          title: proposal.proposal.title,
+          status: proposal.proposal.status as ProposalStatus,
+          votingEndTime: parseDate(proposal.proposal.voting_end_time),
+          depositEndTime: parseDate(proposal.proposal.deposit_end_time),
+          resolvedHeight: proposal.resolved_height,
+          type: proposal.proposal.type as ProposalType,
+          proposer: proposal.proposal.account?.address as Addr,
+        }))
+      );
+  }, [contractAddress, offset, pageSize, indexerGraphClient]);
+
+  return useQuery(
+    [
+      "related_proposals_by_contract_address_pagination",
+      contractAddress,
+      offset,
+      pageSize,
+      indexerGraphClient,
+    ],
+    queryFn,
+    {
+      keepPreviousData: true,
+      enabled: !!contractAddress,
+    }
+  );
+};
+
+export const useRelatedProposalsCountByContractAddress = (
+  contractAddress: ContractAddr
+): UseQueryResult<Option<number>> => {
+  const { indexerGraphClient } = useCelatoneApp();
+
+  const queryFn = useCallback(async () => {
+    return indexerGraphClient
+      .request(getRelatedProposalsCountByContractAddress, {
+        contractAddress,
+      })
+      .then(
+        ({ contract_proposals_aggregate }) =>
+          contract_proposals_aggregate.aggregate?.count
+      );
+  }, [contractAddress, indexerGraphClient]);
+
+  return useQuery(
+    [
+      "related_proposals_count_by_contract_address",
+      contractAddress,
+      indexerGraphClient,
+    ],
+    queryFn,
+    {
+      keepPreviousData: true,
+      enabled: !!contractAddress,
+    }
+  );
+};
+
+export const useProposalsByWalletAddressPagination = (
+  walletAddress: HumanAddr,
+  offset: number,
+  pageSize: number
+): UseQueryResult<UserProposal> => {
+  const { indexerGraphClient } = useCelatoneApp();
+  const queryFn = useCallback(async () => {
+    if (!walletAddress)
+      throw new Error(
+        "Wallet address not found (useProposalsByWalletAddressPagination)"
+      );
+
+    return indexerGraphClient
+      .request(getProposalsByWalletAddressPagination, {
+        walletAddress: walletAddress as HumanAddr,
+        offset,
+        pageSize,
+      })
+      .then(({ proposals }) =>
+        proposals.map((proposal) => ({
+          proposalId: proposal.id,
+          title: proposal.title,
+          status: proposal.status as ProposalStatus,
+          votingEndTime: parseDate(proposal.voting_end_time),
+          depositEndTime: parseDate(proposal.deposit_end_time),
+          type: proposal.type as ProposalType,
+        }))
+      );
+  }, [indexerGraphClient, offset, pageSize, walletAddress]);
+
+  return useQuery(
+    [
+      "proposals_by_wallet_address_pagination",
+      walletAddress,
+      indexerGraphClient,
+      offset,
+      pageSize,
+    ],
+    queryFn,
+    {
+      enabled: !!walletAddress,
+    }
+  );
+};
+
+export const useProposalsCountByWalletAddress = (
+  walletAddress: Option<HumanAddr>
```

Comment:
Is it optional or not? It is not the same as above.

Reply:
Should not be optional